build-logic/conventions/src/main/kotlin/com/squareup/anvil/conventions/BasePlugin.kt

```diff
@@ -98,15 +98,23 @@ abstract class BasePlugin : Plugin<Project> {
         }
 
         freeCompilerArgs.addAll(extension.kotlinCompilerArgs.get())
-        if (extension.explicitApi.get()) {
-          freeCompilerArgs.add("-Xexplicit-api=strict")
+
+        fun isTestOrGradleTest(): Boolean {
+          val sourceSetName = task.sourceSetName
+            .getOrElse(
+              task.name.substringAfter("compile")
+                .substringBefore("Kotlin")
+                .replaceFirstChar(Char::lowercase),
+            )
+
+          return sourceSetName == "test" || sourceSetName == "gradleTest"
         }
 
-        val fromInt = when (val targetInt = target.jvmTargetInt()) {
-          8 -> JVM_1_8
-          else -> JvmTarget.fromTarget("$targetInt")
+        if (extension.explicitApi.get() && !isTestOrGradleTest()) {
```

Comment:
Is the thought here just that applying it for test source sets doesn't really matter and adds unnecessary noise?

Reply:
Exactly!